src/bitcoin/electrum/mod.rs

```diff
@@ -0,0 +1,222 @@
+use std::collections::HashMap;
+
+use bdk_electrum::bdk_chain::{
+    bitcoin::{self, bip32::ChildNumber, BlockHash, OutPoint},
+    local_chain::LocalChain,
+    spk_client::{FullScanRequest, SyncRequest},
+    ChainPosition,
+};
+
+pub mod client;
+mod utils;
+pub mod wallet;
+use crate::bitcoin::{Block, BlockChainTip, Coin};
+
+/// An error in the Electrum interface.
+#[derive(Debug)]
+pub enum ElectrumError {
+    Client(client::Error),
+    GenesisHashMismatch(
+        BlockHash, /*expected hash*/
+        BlockHash, /*server hash*/
+        BlockHash, /*wallet hash*/
+    ),
+}
+
+impl std::fmt::Display for ElectrumError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            ElectrumError::Client(e) => write!(f, "Electrum client error: '{}'.", e),
+            ElectrumError::GenesisHashMismatch(expected, server, wallet) => {
+                write!(
+                    f,
+                    "Genesis hash mismatch. The genesis hash is expected to be '{}'. \
+                    The server has hash '{}' and the wallet has hash '{}'.",
+                    expected, server, wallet,
+                )
+            }
+        }
+    }
+}
+
+/// Interface for Electrum backend.
+pub struct Electrum {
+    client: client::Client,
+    bdk_wallet: wallet::BdkWallet,
+    /// Used for setting the `last_seen` of unconfirmed transactions in a strictly
+    /// increasing manner.
+    sync_count: u64,
+}
+
+impl Electrum {
+    pub fn new(
+        client: client::Client,
+        bdk_wallet: wallet::BdkWallet,
+    ) -> Result<Self, ElectrumError> {
+        Ok(Self {
+            client,
+            bdk_wallet,
+            sync_count: 0,
+        })
+    }
+
+    pub fn sanity_checks(&self, expected_hash: &bitcoin::BlockHash) -> Result<(), ElectrumError> {
+        let server_hash = self.client.genesis_block().hash;
+        let wallet_hash = self.bdk_wallet.local_chain().genesis_hash();
```

Comment:
do we really need `expected_hash` here, as in our single use case of this method, IIUC bdk_wallet.local_chain is previously already created from this `expected_hash`? 
it looks redundant to me

Reply:
I think here it's OK because the `Electrum` object doesn't create `bdk_wallet` itself so doesn't know for sure if it used `expected_hash`.